lib/docker-toolbelt.coffee

```diff
@@ -102,6 +102,44 @@ Docker::imageRootDir = (image) ->
 					else
 						throw new Error("Unsupported driver: #{dockerInfo.Driver}/")
 	)
+
+EEXIST = code: 'EEXIST'
+ignore = ->
+MIN_PAGE_SIZE = 4096
+
+aufsMountWithDisposer = (target, layerDiffPaths) ->
+	# We try to create the target directory.
+	# If it exists, it's *probably* from a previous run of this same function,
+	# and the mount will fail if the directory is not empty or something's already mounted there.
+	fs.mkdirAsync(target)
+	.catch code: EEXIST, ignore
+	.then ->
+		appendFromIndex = null
+		options = 'noxino,ro,br='
+		remainingBytes = MIN_PAGE_SIZE - options.length
+		for i in [0...layerDiffPaths.length]
```

Comment:
I think this entire loop could be replaced by something like
```coffee
[initialLayerPaths, appendLayerPaths] = _(layerDiffPaths)
.map (path) ->
	return "#{path}=ro+wh"
.partition (path) ->
	remainingBytes -= path.length + 1
	# >= -1 because if this is the last entry we won't actually add the comma
	return remainingBytes >= -1
.value()
options += initialLayerPaths.join(':')
...
Promise.mapSeries [appendLayerPaths], (path) ->
	execAsync("mount -t aufs -o 'remount,append:#{path}' none #{target}")
```
(I'm not 100% on this precise version, but I definitely like the `.map (path) -> "#{path}=ro+wh"` to format them all in one pass/one place)